### PR TITLE
Issue #1 set up node and react docker containers

### PR DESCRIPTION
**Pull request by michaelpede**
This will set up docker for local DEV environments and should work for servers as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,57 +4,66 @@ ULI Service with Backend Ingest Process and Matching Service for User Deduplicat
 This work uses the [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) probabilistic matching algorithm for its search implementation, with a set of weights similar to those outlined in the [Sandbox document](https://docs.google.com/document/d/10YFyqw9hIwBXPjpX6yGFQoJUHWpL5M33sVHp5sEjX-Y/edit?usp=sharing) discussed in the RESO ULI Subgroup previously.
 
 
-This repository contains the following items:
```
Comment (darnjo) [inline README.md:7]: We need to keep the outline of what the repo contains.

File: README.md [hunk 2]
```diff
@@ -4,57 +4,66 @@ ULI Service with Backend Ingest Process and Matching Service for User Deduplicat
...
-* A [template spreadsheet](https://github.com/RESOStandards/uli-service/blob/main/ULI%20-%20Data%20Pilot%20Template.xlsx?raw=true), which will allow you to ingest data into the ULI Pilot backend using your local data. 
-* A [`docker-compose.yml` file](https://github.com/RESOStandards/uli-service/blob/main/docker-compose.yml) that will start the Elastic backend for you.
-* A [text file](https://github.com/RESOStandards/uli-service/blob/main/uli-pilot-ingest.txt) containing an Elastic Ingest Pipeline for the template spreadsheet.
-* A file containing a [ULI Pilot Search query](https://github.com/RESOStandards/uli-service/blob/main/uli-ranking-formula.json), which you can fill in values for once you have ingested data.
-
-# Getting Started
-
-## 1. Preparing ULI Pilot Data for Ingest
```
Comment (darnjo) [inline README.md:15]: We still want this section so people can know what to provide. For the initial pilot, an XLSX or CSV may be easier than providing data in JSON format for ingest, as the API requires.

File: README.md [hunk 3]
```diff
@@ -4,57 +4,66 @@ ULI Service with Backend Ingest Process and Matching Service for User Deduplicat
...
-
-The rest of the steps in this README are optional, but to participate in the ULI Pilot there needs to be an initial seed file created from the Member and Office data in the organization at that time. 
-
-A [Template Spreadsheet](https://github.com/RESOStandards/uli-service/blob/main/ULI%20-%20Data%20Pilot%20Template.xlsx?raw=true) has been provided for your convenience. 
-
-Please fill in the fields on the "Merged" tab of the spreadsheet and send the sheet to [RESO Development](mailto:dev@reso.org). Feel free to reach out with any additional questions. 
-
-For those who are running the server locally, please proceed to the next step after filling their data into the Merged sheet.
```
Comment (darnjo) [inline README.md:23]: This is where the instructions for the server should start, and should be contained in a separate `ULI_SERVICE.md` file under a folder called /doc and linked to from a summary section in the main README.

File: README.md [hunk 4]
```diff
@@ -4,57 +4,66 @@ ULI Service with Backend Ingest Process and Matching Service for User Deduplicat
...
+### Setting up project for the first time
 
+Make sure you have [Docker and Docker Compose installed](https://docs.docker.com/compose/install/). The Windows and MacOs installers bundle them both together. The referenced guide has instructions for how to get started with both. 
 
-## 2. Starting the Elastic Backend
+- Change permissions to a sub-folder `sudo chown -R 1000:1000 ./docker-data-volumes/elasticsearch`
+- Run your project with the command `docker-compose up`
```
Comment (darnjo) [inline README.md:12]: Didn't work for me on Linux 5.15.8, Docker version 20.10.7, build 20.10.7-0ubuntu5.1, and Ubuntu 21.10.

Got the following error:

```
docker-compose up
Starting uli_elasticsearch ... error

ERROR: for uli_elasticsearch  Cannot start service elasticsearch: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "/home/jdarnell/work/reso/github/uli-service/elasticsearch.yml" to rootfs at "/usr/share/elasticsearch/config/elasticsearch.yml" caused: mount through procfd: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

ERROR: for elasticsearch  Cannot start service elasticsearch: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "/home/jdarnell/work/reso/github/uli-service/elasticsearch.yml" to rootfs at "/usr/share/elasticsearch/config/elasticsearch.yml" caused: mount through procfd: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
ERROR: Encountered errors while bringing up the project.
```

Stopping here for now until this works.

File: README.md [hunk 5]
```diff
@@ -4,57 +4,66 @@ ULI Service with Backend Ingest Process and Matching Service for User Deduplicat
...
-
-## 4. Querying the Pilot Data
-After you have ingested the data, you can query the server using Kibana's [Dev Tools](http://localhost:5601/app/dev_tools#/console). These will connect to the local Elastic instance, and provides a convenient place to try out queries. 
-
-In this case, you'll want to use something similar to the query that's posted in the [ULI Pilot search endpoint](https://github.com/RESOStandards/uli-service/blob/main/uli-pilot-search.txt):
-
-![dev-tools](https://user-images.githubusercontent.com/535358/121968113-7cbacb80-cd26-11eb-917d-1e5093242e09.png)
+ An example is shown below:
+```
+curl --location --request POST 'http://localhost:3000/api/v1/certification_reports/data_dictionary/M00000595'
```
Comment (darnjo) [inline README.md:32]: Certification example rather than ULI.

File: client/apis/certification.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import axios from "axios";
+
+export const getTermsAndConditions = () =>
+  axios.get(`/api/v1/certification/terms`);
```
Comment (darnjo) [inline client/apis/certification.js:4]: Certification file, etc. All similar files should be removed or stripped down to their bare minimums.

**Review by darnjo — COMMENTED** [review]
See comments so far for changes to be made.

**Comment by darnjo** [comment]
Still getting the following error when trying to use `docker-compose up`:

```
Starting uli_elasticsearch ... error

ERROR: for uli_elasticsearch  Cannot start service elasticsearch: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "/home/jdarnell/work/reso/github/uli-service/elasticsearch.yml" to rootfs at "/usr/share/elasticsearch/config/elasticsearch.yml" caused: mount through procfd: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

ERROR: for elasticsearch  Cannot start service elasticsearch: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "/home/jdarnell/work/reso/github/uli-service/elasticsearch.yml" to rootfs at "/usr/share/elasticsearch/config/elasticsearch.yml" caused: mount through procfd: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
ERROR: Encountered errors while bringing up the project.
```

File: README.md (hunk 4 above)
Comment (michaelpede) [reply README.md:19]: I have:
Docker version 20.10.8
docker-compose version 1.29.2

You may need to upgrade.  Using version: "3.9" for docker-compose needs the bleeding edge of these programs.
I'll update the documentation.

This is exactly how ther reso-certification project works.  Are you using a different environment as your development environment for that project?

Comment (michaelpede) [reply README.md:19]: Ah, I see a file was also missing from the repo.  Now that I've pruned the .p12 files, I can include the needed file.

**Pull request closed without merging**